shap/explainers/_gradient.py

```diff
@@ -291,27 +291,27 @@ def shap_values(self, X, nsamples=200, ranked_outputs=None, output_rank_order="m
                 for k in range(nsamples):
                     rind = np.random.choice(self.data[0].shape[0])
                     t = np.random.uniform()
-                    for l in range(len(X)):
+                    for u in range(len(X)):
```

Comment:
very minor NIT: why use `u` here instead of `i`. I would consider `i` more intuitive for row indices, and `j` for columns.